server/core/basic_cluster.go

```diff
@@ -317,9 +393,43 @@ func (bc *BasicCluster) PreCheckPutRegion(region *RegionInfo) (*RegionInfo, erro
 func (bc *BasicCluster) PutRegion(region *RegionInfo) []*RegionInfo {
 	bc.Lock()
 	defer bc.Unlock()
+	if bc.SelectConf.NewRegionFirst && region.GetTimestamp() >= uint64(time.Now().Unix())-bc.SelectConf.TimeThreshold {
+		bc.NewRegions.SetRegion(region)
+		bc.RemoveOldRegion()
+		bc.CalculateWeights()
```

Comment:
If PutRegion is called frequently, every time we need to scan a lot of regions, which will cost much.

Reply:
I had planned to scan regions periodically, but I did not complete it. Maybe scan is not a good idea. I will maintain this new set in other ways.